doc/source/ray-core/internals/autoscaler-v2.rst

```diff
@@ -0,0 +1,204 @@
+.. _autoscaler-v2:
+
+autoscaler v2
+=============
+
+This document explains how the open-source autoscaler v2 works in Ray 2.48 and outlines its high-level responsibilities and implementation details.
+
+
+Overview
+--------
+
+The autoscaler is responsible for resizing the cluster based on resource demand including tasks, actors, and placement groups.
+To achieve this, it follows a structured process: evaluating worker group configurations, periodically reconciling cluster state with user constraints, applying bin-packing strategies to pending workload demands, and interacting with cloud providers through the Instance Manager.
+The following sections describe these components in detail.
+
+Worker Group Configurations
+---------------------------
+
+Worker groups (also referred to as node types) define the sets of nodes that the Ray autoscaler scales.
+Each worker group represents a logical category of nodes with the same resource configurations, such as CPU, memory, GPU, or custom resources.
+
+The autoscaler dynamically adjusts the cluster size by adding or removing nodes within each group as workload demands change. In other words, it scales the cluster by modifying the number of nodes per worker group according to the specified scaling rules and resource requirements.
+
+Worker groups can be configured in these ways:
+
+- The `available_node_types <https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#node-types>`__ field in the Cluster YAML file, if you are using the ``ray up`` cluster launcher.
```

Comment:
Since this doc is based on 2.48, don't use `latest`, use the 2.48 version of the doc.

Reply:
👍